src/views/Business.vue

```diff
@@ -0,0 +1,121 @@
+        border-gray-600 mb-6 shadow rounded h-64"
+        v-if="hasLocations"
+      >
+        <div class="flex-grow flex-1 flex flex-col pad-6">
+          <div
+            class="flex flex-col"
+
+            v-for="location in business.locations"
+            :class="isSelected(location) ? 'flex-grow' : ''"
+            :key="location.id"
+            @click="() => setSelectedLocation(location)"
+          >
+            <div
+              class="name p-4 border-gray-600 cursor-pointer border-t z-20
+              -m-px transition-colors duration-300 hover:bg-gray-200"
+
+              :class="[ isSelected(location) ? 'border-b border-gray-300 bg-gray-200' : '' ]"
+            >{{location.name}}</div>
+            <div
+              class="border-b border-gray-300 flex flex-row z-1"
+              v-if="isSelected(location)"
+            >
+              <div class="border-r font-bold px-4 p-2">Address:</div>
+              <div class="p-2 text-gray-800 flex-1">
+                {{location.postal_address ? location.postal_address : 'No Postal Address Provided'}}
+              </div>
+            </div>
+          </div>
+        </div>
+        <MglMap
+          class="h-full border-l border-gray-600 flex-grow flex-1 z-30"
+          :accessToken="accessToken"
+          :mapStyle.sync="mapStyle"
+          :center="business.locations[0].location.coordinates"
+          :zoom="9"
+          @load="onMapLoaded"
+        >
+          <MglMarker
+            v-for="location in business.locations"
+            :key="location.id"
+            :coordinates="location.location.coordinates"
+            color="#edf2f7"
+          />
+        </MglMap>
+      </div>
+    </div>
+  </div>
+</template>
+<script>
+import BusinessCard from '@/components/BusinessCard'
+import { GET_BUSINESS_BY_ID } from '@/queries'
+import Mapbox from 'mapbox-gl'
+import { MglMap, MglMarker } from 'vue-mapbox'
+
+export default {
+  props: ['id'],
+  components: { BusinessCard, MglMap, MglMarker },
+  data () {
+    return {
+      accessToken: 'pk.eyJ1Ijoibmlja3NhaGxlciIsImEiOiJja2l5MG9zd2UweWdmMnJwM3pmOTBkbnZlIn0.0PlHofmWvqKzjXi_GBHYZg',
+      mapStyle: 'mapbox://styles/mapbox/streets-v11',
+      selectedLocation: null
+    }
+  },
+  created () { this.mapbox = Mapbox },
+  methods: {
+    setSelectedLocation (location) {
+      this.selectedLocation = location
+      this.flyTo(this.selectedLocation.location.coordinates)
+    },
+    onMapLoaded (event) {
+      this.map = event.map
+      this.flyTo(this.selectedLocation.location.coordinates)
+    },
+    flyTo (center, zoom = 13, speed = 1) {
+      this.map.flyTo({ center, zoom, speed })
+    },
+    isSelected (location) {
+      return this.selectedLocation.id === location.id
+    }
+  },
+  computed: {
+    hasLocations () {
+      return this.business && this.business.locations.length > 0
+    }
+  },
+  watch: {
+    business () {
+      this.setSelectedLocation(this.business.locations[0])
```

Comment:
It might not be apparent to another developer that we only are allowing the user to add at most 1 location so that's why we only care about the first one here. Adding a comment might help somewhere in the code

Reply:
I think I'll add a followup PR with the business-owner views to clean this up - as I'll be poking at data schemas